sources/rest_api/config_setup.py

```diff
@@ -83,15 +89,50 @@ def create_paginator(paginator_config: PaginatorType) -> Optional[BasePaginator]
 
 
 def create_auth(
-    auth_config: Optional[Union[AuthConfig, AuthConfigBase]],
+    auth_config: Optional[Union[SimpleTokenAuthConfig, AuthConfigBase]],
 ) -> Optional[AuthConfigBase]:
     if isinstance(auth_config, AuthConfigBase):
         return auth_config
-    return (
-        BearerTokenAuth(cast(TSecretStrValue, auth_config.get("token")))
-        if auth_config
-        else None
-    )
+
+    if isinstance(auth_config, dict):
+        # Handle a shorthand auth configuration
+        if "token" in auth_config and len(auth_config) == 1:
```

Comment:
@rudolfix I kept this shorthand for convenience but don't have a strong opinion how useful it could be. What do you think?

Reply:
IMO it will be useful when we move to core library. then you'll do
```
for page in requests.paginate(auth="token"):
    ...
 
```
and actual token will come from secrets.toml
```
[sources.name.auth]
api_token="..."
``` 

we need aliases for all auths we support

Reply:
The above works a bit different and is a shorthand for this version:

```py
"client": {
    "base_url": "https://api.github.com/repos/dlt-hub/dlt/",
    "auth": {
        "token": "...",
    },
},
```

For the other auth types I've used the OpenAPI-style definitions, based on what we have in configspecs. E.g.

```py
"client": {
    "base_url": "...",
    "auth": {
        "type": "http",
        "scheme": "bearer",        
    },
},
```

or 

```py
"client": {
    "base_url": "...",
    "auth": {
        "type": "apiKey",
        "name": "X-API-Key",        
         ...
    },
},
```